src/benchmarklib/tpch/tpch_benchmark_item_runner.cpp

```diff
@@ -27,40 +28,50 @@ std::string calculate_date(boost::gregorian::date date, int months, int days = 0
 
 namespace opossum {
 
-TPCHQueryGenerator::TPCHQueryGenerator(bool use_prepared_statements, float scale_factor)
-    : _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
-  _selected_queries.resize(22);
-  std::iota(_selected_queries.begin(), _selected_queries.end(), QueryID{0});
+TPCHBenchmarkItemRunner::TPCHBenchmarkItemRunner(const std::shared_ptr<BenchmarkConfig>& config,bool use_prepared_statements, float scale_factor)
+    : AbstractBenchmarkItemRunner(config), _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
+  _selected_items.resize(22);
+  std::iota(_selected_items.begin(), _selected_items.end(), BenchmarkItemID{0});
+  _prepare_queries();
 }
 
-TPCHQueryGenerator::TPCHQueryGenerator(bool use_prepared_statements, float scale_factor,
-                                       const std::vector<QueryID>& selected_queries)
-    : _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
-  _selected_queries = selected_queries;
+TPCHBenchmarkItemRunner::TPCHBenchmarkItemRunner(const std::shared_ptr<BenchmarkConfig>& config,bool use_prepared_statements, float scale_factor,
+                                                 const std::vector<BenchmarkItemID>& selected_items)
+    : AbstractBenchmarkItemRunner(config), _use_prepared_statements(use_prepared_statements), _scale_factor(scale_factor) {
+  _selected_items = selected_items;
+  _prepare_queries();
```

Comment:
Running queries/doing a lot of work - including optimization (which might even be of benchmarking-interest) - in a constructor seems suboptimal. Why not run the preparation of item X in the first call to `execute_item(X)`?

Reply:
Previously, the preparation was also outside of the benchmark. I'd rather keep execute_item simple.

Reply:
I won't let "previously" count as an argument - this PR is supposed to get us closer to a "proper" benchmarking suite. 


If a benchmark does what I'd regard as substantial work (optimizing those prepared queries) I expect it to report what it did. 
From my personal use case of benchmarking, monitoring optimization is just as important as monitoring execution.

Reply:
>  From my personal use case of benchmarking, monitoring optimization is just as important as monitoring execution.

Which is why preparation is disabled by default.

> If a benchmark does what I'd regard as substantial work (optimizing those prepared queries) I expect it to report what it did.

Hiding it in the first execution of a query doesn't seem like the right way to do it either though. If the TPC-H was a real application, it would also register its prepared statements before they are actually used.